File: CS50-Intro-to-CS/pset1/cash.c
// Cash
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // get user input => how much change is owed
    float change;
    do
    {
        change = get_float("How much change is owed?\n");
    }
    while(change < 0);

    // return how much change is required
    float coins = round(change*100);
    // while(coins > 100)
    // {
    //     coins = coins - 100;
    // }

    // printf("%f", coins);
    // printf("\n");

    int count = 0;
    while(coins >= 25)
    {
        coins = coins - 25;
        count++;
        // printf("QUARTER\n");
        // printf("Coins: %f, Count: %d\n", coins, count);
    }

    while(coins >= 10)
    {
        coins = coins - 10;
        count++;
        // printf("DIME\n");
        // printf("Coins: %f, Count: %d\n", coins, count);
    }
    // printf("Right before nickel");
    while(coins >= 5)
    {
        coins = coins - 5;
        count++;
        // printf("NICKEL\n");
        // printf("Coins: %f, Count: %d\n", coins, count);
    }

    while(coins >= 1)
    {
        coins = coins - 1;
        count++;
        // printf("PENNY\n");
        // printf("Coins: %f, Count: %d\n", coins, count);
    }
    printf("%d\n", count);
}

File: CS50-Intro-to-CS/pset6/sentimental/cash.py
from cs50 import get_int
from cs50 import get_float

while True:
    num = get_float("How much change is owed? ")
    if num > 0:
        break

change = num * 100

coins = 0

while (change >= 25):
    change -= 25
    coins += 1

#print(f"After Quarters: {change} remaining, {coins} required")

while (change  >= 10):
    change -= 10
    coins += 1

#print(f"After Dimes: {change} remaining, {coins} required")

while (change  >= 5):
    change -= 5
    coins += 1

#print(f"After Nickels: {change} remaining, {coins} required")

while (change  >= 1):
    change -= 1
    coins += 1

#print(f"After Pennies: {change} remaining, {coins} required")

print(f"{coins}")